src/clients/python/src/tigerbeetle_python/client.py

```diff
@@ -0,0 +1,226 @@
+from __future__ import annotations
+
+import asyncio
+import ctypes
+import logging
+import os
+import random
+import threading
+import time
+from collections.abc import Callable  # noqa: TCH003
+from dataclasses import dataclass
+from typing import Any
+
+from . import bindings
+
+logger = logging.getLogger("tigerbeetle")
+
+
+class AtomicInteger:
+    def __init__(self, value=0):
+        self._value = value
+        self._lock = threading.Lock()
+
+    def increment(self):
+        with self._lock:
+            self._value += 1
+            return self._value
+
+
+@dataclass
+class CompletionContextSync:
+    event: threading.Event
+
+
+@dataclass
+class CompletionContextAsync:
+    loop: asyncio.AbstractEventLoop
+    event: asyncio.Event
+
+
+@dataclass
+class InflightPacket:
+    packet: bindings.CPacket
+    response: Any
+    operation: bindings.Operation
+    c_event_type: Any
+    c_result_type: Any
+    on_completion: Callable | None
+    on_completion_context: CompletionContextSync | CompletionContextAsync | None
+
+
+
+def tbid() -> int:
```

Comment:
I dropped the strictly monotonic increasing timestamp parts from the Node client... I'm not sure they're worth it? I can put them back.

Reply:
Maybe it is worth it? If time goes backwards due to ntp adjustment or what not, we will get a latency spike as we'd have to go all the way to the bottom of LSM to disconfirm transfer existence.